src/openloco/windows/station_list.cpp

```diff
@@ -49,4 +184,163 @@ namespace openloco::ui::windows::station_list
 
         return station_list;
     }
+
+    static void event_08(window* window)
+    {
+        window->flags |= window_flags::flag_14;
+    }
+
+    static void event_09(window* window)
+    {
+        if ((window->flags & window_flags::flag_14) == 0)
+            return;
+
+        if (window->row_hover == 0xFFFF)
+            return;
+
+        window->row_hover = -1;
+        window->invalidate();
+    }
+
+    // 0x00491344
+    static void prepare_draw(ui::window* window)
+    {
+        registers regs;
+        regs.esi = (int32_t)window;
+        call(0x00491344, regs);
+    }
+
+    // 0x004914D8
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        // // Draw widgets.
+        window->draw(dpi);
+
+        registers regs;
+        regs.esi = (int32_t)window;
+        regs.edi = (int32_t)dpi;
+
+        // This sub is only used in this drawing routine. Integrate.
+        call(0x00491A76, regs);
+
+        // Continue drawing with the original routine.
+        call(0x004914E2, regs);
+    }
+
+    // 0x004917BB
+    static void on_dropdown(ui::window* window, widget_index widgetIndex, int16_t itemIndex)
+    {
+        if (widgetIndex == widx::company_select)
+        {
+            registers regs;
+            regs.edx = (int32_t)widgetIndex;
+            regs.esi = (int32_t)window;
+            regs.ax = itemIndex;
+
+            call(0x004917C2);
+        }
+    }
+
+    // 0x004917B0
+    static void on_mouse_down(ui::window* window, widget_index widgetIndex)
+    {
+        if (widgetIndex == widx::company_select)
+        {
+            registers regs;
+            regs.edx = widgetIndex;
+            regs.edi = (int32_t)&_widgets[widgetIndex];
```

Comment:
Still draws the dropdown in the wrong position.

Reply:
```suggestion
            regs.edi = (int32_t)&window->widgets[widgetIndex];
```

(In prepare_draw, window->widgets is being overwritten.)

Reply:
Ah. It's literally the same widgets for each tab, though, so we can skip that when implementing this function.